src/AstTransforms.cpp

```diff
@@ -1414,4 +1414,289 @@ bool AstUserDefinedFunctorsTransformer::transform(AstTranslationUnit& translatio
     return update.changed;
 }
 
+bool FoldAnonymousRecords::isValidRecordConstraint(const AstLiteral* literal) {
+    auto constraint = dynamic_cast<const AstBinaryConstraint*>(literal);
+
+    if (constraint == nullptr) {
+        return false;
+    }
+
+    const auto* left = constraint->getLHS();
+    const auto* right = constraint->getRHS();
+
+    const auto* leftRecord = dynamic_cast<const AstRecordInit*>(left);
+    const auto* rightRecord = dynamic_cast<const AstRecordInit*>(right);
+
+    // Check if arguments are records records.
+    if ((leftRecord == nullptr) || (rightRecord == nullptr)) {
+        return false;
+    }
+
+    // Check if records are of the same size.
+    if (leftRecord->getChildNodes().size() != rightRecord->getChildNodes().size()) {
+        return false;
+    }
+
+    // Check if operator is "=" or "!="
+    auto op = constraint->getOperator();
+    if (!isEqConstraint(op) && !isEqConstraint(negatedConstraintOp(op))) {
+        return false;
+    }
+
+    return true;
```

Comment:
The linter would like this to be
```
    return !(!isEqConstraint(op) && !isEqConstraint(negatedConstraintOp(op)));
```
the simpler version seems preferable
```suggestion
    return isEqConstraint(op) || isEqConstraint(negatedConstraintOp(op));
```